### ISSUE TITLE
[Bug]: Output segmentation mask mismatch

### ISSUE THREAD

**Issue by 1713mz** [open-edge-platform/anomalib#1660]
### Describe the bug

Hello, I have recently found that in the output images the subplots of image/predicted heatmap and groundtruth/predicted mask/segmentation result does not match, see the following image:
![003](https://github.com/openvinotoolkit/anomalib/assets/98899153/009b1afd-31f8-4953-b03c-f11ec97757f3)
I am pretty sure it was caused by the centercrop operation. After disabling the parameter the results looks reasonable:
![003_new](https://github.com/openvinotoolkit/anomalib/assets/98899153/a0802fe7-629d-4396-b7e1-f0231eb7e020)

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

None

### OS information

Not relevant

### Expected behavior

None

### Pip/GitHub

GitHub

### Configuration YAML

```yaml
Not relevant
```

### Logs

```shell
Not relevant
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
Hi, thanks for reporting this. It looks very similar to #1447 so I think there is actually a problem with center cropping.
I think the problem arises because inside visualizer the image is loaded again, since the original was normalized:
https://github.com/openvinotoolkit/anomalib/blob/9effa29583e00dc814e948b60d255b8ecbd5dca3/src/anomalib/post_processing/visualizer.py#L102-L103

I belive this needs to be handled differently. Either we also pass center crop data to visualizer, or we pass the normalization data to de-normalize the image.
Looking at the history, it seems like there used to be a denormalization which was removed due to an issue:
https://github.com/openvinotoolkit/anomalib/blob/4860abc78b0976faf2c423713d5b90cfaa55ed22/anomalib/post_processing/visualizer.py#L77-L78

@samet-akcay do you maybe remember why that was the case and what should be done in this case since visualizing with center-crop seems to be an issue?

**Comment by samet-akcay** [comment]
Yeah, I think it was to avoid denormalization if i remember correctly, @djdameln? 

@ashwinvaidya17 is currently working on #1530. Maybe we could consider addressing it there, or a separate one?

**Comment by Akash-Chokhani** [comment]
Hi, I would like to work on this issue.

**Comment by blaz-r** [comment]
Sure 😊, thanks.

**Comment by samet-akcay** [comment]
Thanks @Akash-Chokhani !

**Comment by Akash-Chokhani** [comment]
Got it. Thanks for letting me know.

**Comment by samet-akcay** [comment]
We will be able to address this issue with https://github.com/openvinotoolkit/anomalib/pull/2358 and https://github.com/openvinotoolkit/anomalib/pull/2334

**Comment by watertianyi** [comment]
> centercrop

@ashwinvaidya17 @samet-akcay How is this solved?

**Comment by mzweilin** [comment]
PR https://github.com/openvinotoolkit/anomalib/pull/2063 might solve the issue
